pkg/lockfile/match-build-gradle.go

```diff
@@ -0,0 +1,80 @@
+package lockfile
+
+import (
+	"io"
+	"strings"
+
+	"github.com/google/osv-scanner/internal/utility/fileposition"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+type BuildGradleMatcher struct{}
+
+func (m BuildGradleMatcher) GetSourceFile(lockfile DepFile) (DepFile, error) {
+	fileName := "build.gradle"
+
+	// lockfile (default, groovy)
+	sourcefile, err := lockfile.Open(fileName)
+	if err != nil {
+		// kotlin
+		sourcefile, err = lockfile.Open(fileName + ".kts")
+	}
+
+	// gradle verification metadata (<rootdir>/gradle/verification-metadata.xml)
+	relativePath := "../" + fileName
```

Comment:
Is this a gradle constraint ?

Reply:
When the `verification-metadata.xml` is generated, is always inside a folder named `gradle`. This means that the `build.gradle` file (source file that we are looking for) relative path will be "../build.gradle":

- <rootdir>/
   - gradle/
      - verification-metadata.xml
   - build.gradle
   - gradle.lockfile